lib/parsers/index.ts

```diff
@@ -210,19 +210,36 @@ async function getDependenciesFromReferenceInclude(manifestFile, includeDev: boo
   return referenceIncludeResult;
 }
 
-function buildSubTreeFromPackageReference(dep, isDev: boolean): PkgTree {
+function buildSubTreeFromPackageReference(dep, isDev: boolean, manifestFile): PkgTree {
 
   const depSubTree: PkgTree = {
     depType: isDev ? DepType.dev : DepType.prod,
     dependencies: {},
     name: dep.$.Include,
     // Version could be in attributes or as child node.
-    version: dep.$.Version || _.get(dep, 'Version.0'),
+    version: extractDependencyVersion(dep, manifestFile),
   };
 
   return depSubTree;
 }
 
+function extractDependencyVersion(dep, manifestFile) {
+  const VARS_MATCHER = /^\$\((.*?)\)/;
+  const version = dep.$.Version || _.get(dep, 'Version.0');
```

Comment:
is it ever possible for this to be undefined as in `dep.$.Version` and `_.get(dep, 'Version.0')` are not set?

Reply:
yes it is a possibility, so next step should check for that:` version && version.match(VARS_MATCHER);`

So if no version the expression is `false`

Reply:
cool - I was asking as we return `version` if the version expression is false. Making sure we're okay with returning a null version.